core/includes/date.inc

```diff
@@ -1114,7 +1131,7 @@ function date_has_time($granularity) {
  * @return bool
  *   TRUE if the granularity contains a date portion, FALSE otherwise.
  */
-function date_has_date($granularity) {
+function date_has_date(array $granularity) {
   if (!is_array($granularity)) {
```

Comment:
Same problem as above. This is_array() won't be useful anymore, when using the type hint.